### PR TITLE
[FIX] mass_mailing: properly disable wrong link color option

### PR DESCRIPTION
**Pull request by Zynton**
The background color snippet option contains a superfluous selector. To make sure the option doesn't actually appear for that selector, without modifying the XML file in stable, commit [1] hid it in those cases. But we want the option not to even be instantiated, which this commit achieves by modifying the selector when computing the snippet templates.

This reverts commit [2] so as to avoid updating the DOM in two separate places of the code for link edition. Said commit intended to prevent the removal of custom styles on a button on which styles were applied (erroneously) without applying a custom button class. This fixes that issue by adapting the UI to show the button is custom if there are custom styles applied. This way, if any change is applied, when modifying the DOM, the custom class will be applied as well. This is the order in which such corrections are usually applied with the snippets bar.

[1]: https://github.com/odoo/odoo/commit/925e73e02c5bc054d06fc2097acc9e4244cabe38
[2]: https://github.com/odoo/odoo/commit/eb4edac560227b46efdcfe958e3b93e0b88def64

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by dmo-odoo** [comment]
This is the followup to the background color conundrum of last time. We have implemented the solution that was discussed with @qsm-odoo. What do you think ? cc @bso-odoo

File: addons/web_editor/static/src/js/wysiwyg/widgets/link_tools.js [hunk 1]
```diff
@@ -44,14 +44,24 @@ const LinkTools = Link.extend({
...
         }
         this._addHintClasses();
-        return this._super(...arguments);
+        const ret = await this._super(...arguments);
+        const link = this.$link[0];
+        const customStyleProps = ['color', 'background-color', 'background-image', 'border-width', 'border-style', 'border-color'];
+        if (customStyleProps.some(s => link.style[s])) {
+            // Force custom style if style exists on the link.
+            const options = this.el.querySelectorAll('[name="link_style_color"] we-button');
+            const customOption = [...options].find(opt => opt.dataset.value === 'custom');
```
Comment (qsm-odoo) [inline addons/web_editor/static/src/js/wysiwyg/widgets/link_tools.js:60]: You seem to search all buttons to then find only one inside. Is not this equal to:
```suggestion
            const customOption = this.el.querySelector('[name="link_style_color"] we-button[data-value="custom"]');
```
?

**Review by qsm-odoo — CHANGES_REQUESTED** [review]
Sorry for not having had a look at this sooner. Just a small comment, afterwards I'll r+ tomorrow :+1:

File: addons/web_editor/static/src/js/wysiwyg/widgets/link_tools.js (hunk 1 above)
Comment (Zinston) [reply addons/web_editor/static/src/js/wysiwyg/widgets/link_tools.js:60]: You're absolutely right of course :facepalm: Thanks.

**Review by qsm-odoo — APPROVED** [review]
@robodoo rebase-ff r+

**Pull request closed without merging**